### DISCUSSION TITLE
Huawei Integration for HA at the same time as Modbus Integration for HA

### DISCUSSION THREAD

**Discussion by richardpub4136** [wlcrs/huawei_solar#798] [Connecting to the inverter]
Can I use Huawei Integration for HA at the same time as Modbus Integration for HA, without going through a Modbus Proxy?

**Comment by mad-jack** [comment]
Why did you need a modbus proxy? You can use that integration independently from the native modbus HA integration. I suppose   huawei_solar has an internal and independent modbus management so you don't need to configure a specific  modbus connection for huawei_solar.

**Reply by richardpub4136** [reply]
I use the Huawei solar integration but I am trying to connect to modbus with Node-red but the connection does not connect to my Huawei inverter for which I want to discover the different parameters in real time. I wonder if the number of simultaneous has a limit ???
thanks for tour response

**Reply by crazyfx1** [reply]
Yes you need the modbus proxy if you want to connect multiple systems to the same inverter. Both the HA integration and Node Red have to connect to the proxy.

**Reply by richardpub4136** [reply]
Ok 
Thank you now I know how to proceed to make several connections work at the same time

**Reply by Roving-Ronin** [reply]
Limitation of the connection method, i.e. sDongle only supports a single connection. Believe the new EMMA may support multiple connections though. Luckily with v1.4.x you have reconfigure this integration to point to a new IP (i.e. Modbus Proxy) without needing to uninstall and reinstall.
